### PR TITLE
Refactor Conda Recipes

### PR DESCRIPTION
**Pull request by milljm**
Refactor the recipes to be more loose with dependencies in some areas,
and more strict in others. Namely the compilers.

moose-tools no longer depends on any other moose- package. It is our
hope that cutting ties with moose-mpich will help when users install
other packages.

moose-mpich has a _very_ strict dependency set (compilers, hdf5, mesa).

[edit]:
moose-libmesh-vtk; remove the Qt/PyQt dependency. This will sever ties
with many dependencies that was hindering additional package installs.
This also affords us a single moose-libmesh-vtk build.

moose- petsc, libmesh-vtk, and libmesh no longer depend on a hash_build_#
These packages will now pull whatever is the latest version and available
at the time. After contemplating how our build system works, this should
just always get the right package, locally built, or /public. I believe
it may also be possible to remove the version entirely (do that later).

Add moose-gperftools [edit]: renamed to moose-pprof
Add moose-autojump

In order for all this to work, we had to break away from a common
symlink to conda_build_config.yaml. There remains a few places where we
can still have a common config, but I opted to move forward with
separating everything.

One caveat, is the necessity to start over. No one will be able to
upgrade to any of these packages which depend on moose-mpich. Something
about Python/VTK/PyQt's deps. Once a person has the latest moose-mpich
package, things will start to fall into place.

Refs: #21247 #21111

### CODE REVIEW

File: conda/libmesh-vtk/build.sh [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
+export HYDRA_LAUNCHER=fork
 
 # Qt is enabled in VTK by default, but currently doesn't exist on M1 Macs.
 # So, it is disabled if building on that platform.
 QT="YES"
 if [[ $(uname) == Darwin ]] && [[ $(uname -p) == arm ]]; then
   QT="NO"
 fi
-
+echo "CC, CXX $CC $CXX"
```
Comment (milljm) [inline conda/libmesh-vtk/build.sh:12]: remove this

File: conda/gperftools/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+BSD 3-clause license
+Copyright (c) 2015-2018, conda-forge
+All rights reserved.
```
Comment (cticenhour) [inline conda/gperftools/LICENSE.txt:3]: So we have to include a `conda-forge` license with a Google open source product? Isn't there a Google license we need to use?

File: conda/gperftools/build.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -ex
+
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* .
+if [ `uname` == "Darwin" ]; then
+    TEMP_CXXFLAGS=${CXXFLAGS//-std=c++[0-9][0-9]}
+    # gperftools uses random_shuffle (removed in c++17)
+    ACTIVATION_CXXFLAGS=${TEMP_CXXFLAGS%%-fdebug-prefix-map*}-std=c++14
```
Comment (cticenhour) [inline conda/gperftools/build.sh:10]: Kinda glad this little shuffle is useful elsewhere 😂

File: conda/gperftools/info/has_prefix [hunk 4]
```diff
@@ -0,0 +1 @@
+/opt/anaconda1anaconda2anaconda3 text etc/activate.d/activate_moose-gperftools.sh
```
Comment (cticenhour) [inline conda/gperftools/info/has_prefix:1]: Can you explain what's going on here?

File: conda/mpich/conda_build_config.yaml [hunk 5]
```diff
@@ -1,245 +1,63 @@
...
-## changes made will require additional changes to items listed above
-## your change (except where otherwise noted)
-
-# NCRC dependency
-moose_tools:
-  - moose-tools 2022.04.18
-
-# NCRC default Python
-moose_ncrc_python:
-  - python 3.9
```
Comment (cticenhour) [inline conda/mpich/conda_build_config.yaml:11]: Removing this won't affect anything? I don't know anything about NCRC needs.

File: conda/petsc/run_test.sh [hunk 6]
```diff
@@ -1,6 +0,0 @@
-#!/bin/bash
-set -e
-
-export PETSC_DIR=${PREFIX}
-
-pkg-config --cflags PETSc | grep -v isystem
```
Comment (cticenhour) [inline conda/petsc/run_test.sh:6]: Just double-checking that this is no longer needed. I don't see any changes in the `meta.yaml` regarding testing, so I am assuming we should have removed this a while back?

**Review by cticenhour — COMMENTED** [review]
The GitHub UI shows that `conda/pyhit/build.sh` is an empty file that somehow changes size. When I click on "View File", it takes me to your refactor commit (or at least a version of it) that has content but claims to not be on any branches.....can you double-check that this file exists on your local branch?

File: conda/gperftools/LICENSE.txt (hunk 2 above)
Comment (milljm) [reply conda/gperftools/LICENSE.txt:3]: hmm yeah probably. I'll be sure to figure that out

File: conda/gperftools/info/has_prefix (hunk 4 above)
Comment (milljm) [reply conda/gperftools/info/has_prefix:1]: This is a post process operation that alters any paths it sees in the specified file path (`etc/activate.d/activate_moose-gperftools.sh`), having a prefix with all those:
```pre
_placehold_placehold_placehold_placehold*
```
paths, changed to the correct final path of $CONDA_PREFIX. This is so when all is said and done, the end user john_doe who installed Conda to some location, has a working activation script:

```bash
/Users/john_doe/libs/conda3/envs/moose/etc/conda/activate.d/activate_moose-gperftools.sh
export GPERF_DIR=/Users/john_doe/libs/conda3/envs/moose
```
Since we don't know what that will be before hand, this is basically how Conda does it :). I probably should just have linked the following. Will explain better than I can:
Ref: https://docs.conda.io/projects/conda-build/en/latest/resources/make-relocatable.html

File: conda/petsc/run_test.sh (hunk 6 above)
Comment (milljm) [reply conda/petsc/run_test.sh:6]: Oh, yeah we can probably put this back. :/ not sure how/why I deleted it.

File: conda/gperftools/LICENSE.txt (hunk 2 above)
Comment (cticenhour) [reply conda/gperftools/LICENSE.txt]: So we're using their recipe you mean, and then overwriting? In that case I could see providing both open source licenses in a single file.

File: conda/gperftools/info/has_prefix (hunk 4 above)
Comment (cticenhour) [reply conda/gperftools/info/has_prefix]: Looks good - thanks for the explanation!

**Review by loganharbour — DISMISSED** [review]

File: conda/pprof/LICENSE [hunk 7]
```diff
@@ -0,0 +1,13 @@
+BSD 3-clause license
+Copyright (c) 2015-2018, conda-forge
```
Comment (cticenhour) [inline conda/pprof/LICENSE:2]: OK - I like the move to pprof here....So we've moved to using conda forge gperftools and then downloading and building pprof? Shouldn't this then be the google license?

File: conda/libmesh-vtk/build.sh [hunk 8]
```diff
@@ -27,41 +12,24 @@ cmake .. -G "Ninja" \
     -DCMAKE_PREFIX_PATH:PATH=${VTK_PREFIX} \
     -DCMAKE_INSTALL_PREFIX:PATH=${VTK_PREFIX} \
     -DCMAKE_INSTALL_RPATH:PATH=${VTK_PREFIX}/lib \
-    -DCMAKE_INSTALL_LIBDIR=lib \
```
Comment (cticenhour) [inline conda/libmesh-vtk/build.sh:30]: In your commit message changing the libmesh-vtk package tests in meta.yaml, you wondered why lib needed to be changed to lib64. I think this is why. Why did you remove this?

**Review by cticenhour — DISMISSED** [review]

**Review by loganharbour — DISMISSED** [review]

**Review by cticenhour — DISMISSED** [review]

**Review by cticenhour — APPROVED** [review]
